pkg/controller/cluster/main-controller.go

```diff
@@ -400,19 +439,18 @@ func (c *Controller) Start(threadiness int, stopCh <-chan struct{}) error {
 
 		go func() {
 			klog.Infof("Starting HTTP Upgrade Tenant Image server")
-			close(upgradeServerWillStart)
+			wg.Done()
```

Comment:
better to `defer wg.Done()`

Reply:
same, stays for now, will be refactored on another PR